lib/takeaway.rb

```diff
@@ -0,0 +1,38 @@
+require 'menu'
+require 'basket'
+
+class Takeaway
+  attr_reader :menu
+  attr_reader :basket
+  
+  def initialize(menu_class = Menu, basket = Basket.new)
+    @menu_class = menu_class
+    @menu = nil
+    @basket = basket
+  end
+
+  def view_menu
+    @menu = @menu_class.new
```

Comment:
This line sort of implies that every time I want to look at the menu a new one should be created. That's probably ok here, but it feels to me that it would make more sense for the menu to be created once when the `Takeaway` object is instantiated and then be held in an instance variable.